project_01.py

```diff
@@ -54,8 +54,11 @@ def mkMux(inputs, outputs):
 def mkDMux(inputs, outputs):
     in_ = inputs.in_
     sel = inputs.sel
-    outputs.a = And(a=in_, b=Not(in_=sel).out).out
-    outputs.b = And(a=in_, b=sel).out
+    # This optimal version was found by nand.optimize.super_optimize():
+    nand0 = Nand(a=in_, b=sel).out
+    nand1 = Nand(a=in_, b=nand0).out
```

Comment:
Seems like there might be a somewhat general rewrite rule in here: if `n1 = Nand(x, y)` is already present, then `Nand(Not(x), z)` is equivalent to `Nand(n1, z)`, and the latter saves one gate. Or something like that.